spec/API_specification/array_api/fourier_transform_functions.py

```diff
@@ -0,0 +1,468 @@
+    norm: Literal['backward', 'ortho', 'forward']
+        specify the normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: normalize by ``1/n``.
+        - ``'ortho'``: normalize by ``1/sqrt(n)`` (i.e., make the FFT orthonormal).
+        - ``'forward'``: no normalization.
+
+        where ``n`` equals ``prod(s)``, the logical FFT size.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axes (dimension) indicated by ``axes``. The returned array must have a complex floating-point data type determine by :ref:`type-promotion`.
+    """
+
+
+def rfft(x: array, /, *, n: Optional[int] = None, axis: int = -1, norm: Literal['backward', 'ortho', 'forward'] = 'backward') -> array:
+    """
+    Computes the one-dimensional discrete Fourier transform for real-valued input.
+
+
+    .. note::
+       Applying the one-dimensional inverse discrete Fourier transform for real-valued input to the output of this function must return the original (i.e., non-transformed) input array within numerical accuracy (i.e., ``irfft(rfft(x), n=x.shape[axis]) == x``), provided that the transform and inverse transform are performed with the same normalization mode..
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a real-valued floating-point data type.
+    n: int
+        length of the transformed axis of the **input**. If
+
+        - ``n`` is greater than the length of the input array, the input array must be zero-padded such that the input array has length ``n``.
+        - ``n`` is less than the length of the input array, the input array must be trimmed to length ``n``.
+
+        If not provided, the length of the transformed axis of the output must equal the length of the input along the axis specified by ``axis``. Default: ``None``.
+    axis: int
+        axis (dimension) over which to compute the Fourier transform. If set to ``-1``, the function must compute the Fourier transform over the last axis (dimension). Default: ``-1``.
+    norm: Literal['backward', 'ortho', 'forward']
+        normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: no normalization.
+        - ``'ortho'``: normalize by ``1/sqrt(n)`` (i.e., make the FFT orthonormal).
+        - ``'forward'``: normalize by ``1/n``.
+
+        where ``n`` equals ``prod(s)``, the logical FFT size.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axis (dimension) indicated by ``axis``. The returned array must have a complex floating-point data type determined by :ref:`type-promotion`.
+    """
+
+
+def irfft(x: array, /, *, n: Optional[int] = None, axis: int = -1, norm: Literal['backward', 'ortho', 'forward'] = 'backward') -> array:
+    """
+    Computes the one-dimensional inverse of ``rfft``.
```

Comment:
ditto, C2R transforms (`irfft`, `irfftn`) must take complex input and reject real ones.